lib/src/errors/dson_exception.dart

```diff
@@ -28,5 +32,136 @@ class ParamsNotAllowed extends DSONException {
   ParamsNotAllowed(super.message, [super.stackTrace]);
 
   @override
-  String get _className => 'ParamsNotAllowed';
+  String get _className => '[ParamsNotAllowed]';
+}
+
+/// Called when param is unknown and the library is not able to handle it
+class ParamUnknown extends DSONException {
+  /// The name of the class that contains the param
+  final String? parentClass;
+
+  /// Param name
+  final String? paramName;
+
+  /// Called when param is unknown and the library is not able to handle it
+  ParamUnknown({
+    this.parentClass,
+    this.paramName,
+    StackTrace? stackTrace,
+  }) : super(
+          "Unknown error while trying parse parameter '$paramName' on class"
+          " '$parentClass'",
+          stackTrace,
+        );
+
+  @override
+  String get _className => '[ParamUnknown]';
+}
+
+/// Called when value is null, but params is required and non-nullable
+class ParamNullNotAllowed extends DSONException {
+  /// the representation of the param
+  final FunctionParam functionParam;
+
+  /// The name of the class that contains the param
+  final String parentClass;
+
+  /// Called when value is null, but params is required and non-nullable
+  ParamNullNotAllowed({
+    required this.functionParam,
+    StackTrace? stackTrace,
+    required this.parentClass,
+  }) : super(
+          "Param '${functionParam.name}' from $parentClass"
+          '({required $functionParam})'
+          "${functionParam.alias != null ? " with alias"
+              " '${functionParam.alias}'," : ''}"
+          ' is required and non-nullable, but the value is null or some alias'
+          ' is missing.',
+          stackTrace,
+        );
+
+  @override
+  String get _className => '[ParamNullNotAllowed]';
```

Comment:
É interessante usar uma interpolação com o tipo da classe. Dessa forma, se o nome da classe for alterado essa string será considerada como referência da classe e também será alterada.
```suggestion
  String get _className => '[$ParamNullNotAllowed]';
```

Reply:
done e03bbfcf12b3910cf2a821e421f775da30144206